api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -31,29 +31,31 @@
  *      Additional reviews and feedback by Tomas Langer.
  *
  *******************************************************************************/
-
 package org.eclipse.microprofile.config;
 
 import java.lang.reflect.Array;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
 
-import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigValue;
+import org.eclipse.microprofile.config.spi.ConfigValueSource;
 import org.eclipse.microprofile.config.spi.Converter;
 
 /**
+ * Resolves the property value by searching through all configured {@link ConfigValueSource ConfigSources}. If the same
+ * property is specified in multiple {@link ConfigValueSource ConfigSources}, the value in the
+ * {@link ConfigValueSource} with the highest ordinal will be used.
  * <p>
- * Resolves the property value by searching through all configured
- * {@link ConfigSource ConfigSources}. If the same property is specified in multiple
- * {@link ConfigSource ConfigSources}, the value in the {@link ConfigSource} with the highest
- * ordinal will be used.
- * <p>If multiple {@link ConfigSource ConfigSources} are specified with
- * the same ordinal, the {@link ConfigSource#getName()} will be used for sorting.
+ *
+ * If multiple {@link ConfigValueSource ConfigSources} are specified with the same ordinal, the
```

Comment:
Since we have ConfigSource and ConfigValueSource, I don't think you should remove the mentioning of ConfigSource. What will happen if the ConfigValueSource and ConfigSource have the same ordinal, which one wins? This needs to be speced.

Reply:
It would be exactly the same as having two old style config sources with the same ordinal. It's not a special case.

Reply:
It needs to be speced and ConfigSource mention should not be removed.

Reply:
Idea was to wrap the old ConfigSource in the new ConfigValueSource, so we wouldn't need to refer to both of them all the time.

Reply:
I think it is much simpler and easier to add methods to ConfigSource.

Reply:
As default methods or just abstract methods?

Reply:
abstract methods

Reply:
Seems more reasonable, since we want to force implementations to do it, but also it means we are going to break everyone :D

Reply:
This is why we release Config 2.0 not Config 1.5. We have to go one way or the other. If we lean more for customers to implements the new method, we need to make it abstract unless we can sufficiently implement it.

Reply:
I'm OK with this. Is just that we were trying to prevent it and in the end we just came back to the same place :)